src/libraries/Microsoft.PowerFx.Core/Public/Config/PowerFxConfig.cs

```diff
@@ -48,6 +48,22 @@ public PowerFxConfig(CultureInfo cultureInfo = null)
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PowerFxConfig"/> class.
+        /// Copy constructor. Should only be used on a locked PowerFxConfig object. 
+        /// </summary>
+        /// <param name="other">Config to clone from.</param>
+        private PowerFxConfig(PowerFxConfig other)
+        {
+            _isLocked = other._isLocked;
+            _extraFunctions = other._extraFunctions;
+            _environmentSymbols = other._environmentSymbols;
+            _environmentSymbolDisplayNameProvider = other._environmentSymbolDisplayNameProvider;
+            _coreFunctions = other._coreFunctions;
+            EnumStoreBuilder = other.EnumStoreBuilder;
```

Comment:
This does present a problem. There is special logic in PowerFxConfig that attempts to enforce the following axiom: Every time you add a function to the config, it updates the EnumStoreBuilder by calling WithRequiredEnums. This means that if two configs point to the same builder, they could each add functions. Collisions are impossible, but it could lead to unwanted enums. I am not sure if this is a problem in practice.

Reply:
Right, totally agree. For now, not an immediate issue since this is only being called from a path where it's already locked, but #276 is the right direction here, as you brought up a week ago.